src/simtools/db/db_handler.py

```diff
@@ -364,154 +400,87 @@ def read_mongo_db(
+        collection = self.get_collection(self._get_db_name(), "production_tables")
+        post = collection.find_one(query)
+        if not post:
+            raise ValueError(f"The following query returned zero results: {query}")
+
+        return {
+            "collection": post["collection"],
+            "model_version": post["model_version"],
+            "parameters": post["parameters"],
+            "design_model": post.get("design_model", {}),
+            "entry_date": ObjectId(post["_id"]).generation_time,
         }
-        if only_applicable:
-            query["applicable"] = True
-        if collection.count_documents(query) < 1:
-            raise ValueError(
-                "The following query returned zero results! Check the input data and rerun.\n",
-                query,
-            )
-        for post in collection.find(query).sort("parameter", ASCENDING):
-            par_now = post["parameter"]
-            _parameters[par_now] = post
-            _parameters[par_now].pop("parameter", None)
-            _parameters[par_now].pop("site", None)
-            _parameters[par_now]["entry_date"] = ObjectId(post["_id"]).generation_time
 
-        return _parameters
-
-    def get_derived_values(self, site, array_element_name, model_version):
+    def get_array_elements_of_type(self, array_element_type, model_version, collection):
         """
-        Get all derived values from the DB for a specific array element.
+        Get array elements of a certain type (e.g. 'LSTN') for a DB collection.
 
         Parameters
         ----------
-        site: str
-            Site name.
-        array_element_name: str
-            Name of the array element model (e.g. MSTN, SSTS).
+        array_element_type: str
+            Type of the array element (e.g. LSTN, MSTS).
         model_version: str
             Version of the model.
+        collection: str
+            Which collection to get the array elements from:
+            i.e. telescopes, calibration_devices.
 
         Returns
         -------
-        dict containing the parameters
-
-        """
-        _, _array_element_name, _model_version = self._validate_model_input(
-            site, array_element_name, model_version
-        )
-
-        return self.read_mongo_db(
-            DatabaseHandler.DB_DERIVED_VALUES,
-            _array_element_name,
-            _model_version,
-            run_location=None,
-            collection_name="derived_values",
-            write_files=False,
+        list
+            Sorted list of all array element names found in collection
+        """
+        production_table = self._read_production_table_from_mongo_db(collection, model_version)
+        all_array_elements = production_table["parameters"]
+        return sorted(
+            [
+                entry
+                for entry in all_array_elements
+                if entry.startswith(array_element_type) and "-design" not in entry
```

Comment:
Might be useful to mention in the docstring the fact this skips the design (I wouldn't necessarily expect that).